rfcs/20200106-prelu/README.md

```diff
@@ -0,0 +1,76 @@
+# Introducing PReLU support in oneDNN
+
+## Introduction
+
+This paper https://arxiv.org/abs/1502.01852 introduced PReLU activation function.
+Request comes from Caffe users: "PReLU is used in some POR topologies, such as
+VNET, SRGAN. It takes most of time if the original implementation, is not good
+optimized. For example VNET's train perf with latest Intel Caffe is 2.4s, PReLU
+(has optimized by openmp) spends 429ms (not including reorder time), 17% of total
+time. If we uses oneDNN ReLU to make projection of the perf, it will come to
+~1.8s." Some other data scientist in discussion mentioned that he encouter PReLU
+issue when he tried to optimize the inference pipeline of a popular face-related
+application - PReLU tooks 47 % of time in his application.
+
+Currently in oneDNN we support ReLU activation function in eltwise operation:
+https://oneapi-src.github.io/oneDNN/dev_guide_eltwise.html. LeakyReLU can be
+achieved by specifying non zero alpha parameter. PReLU is specific variant of
+LeakyReLU, where alpha is learnable parameter. It cames with two flavours:
+- channel-wise PReLU - where number of trained alpha parameters is equal to
+channel number
+- channel-shared PReLU - where there is only one trainable alpha
+
+All popular framworks supports PReLU:
+- Caffe - https://caffe.berkeleyvision.org/tutorial/layers/prelu.html
+- Keras/tensorflow - https://www.tensorflow.org/api_docs/python/tf/keras/layers/PReLU
+- PyTorch - https://pytorch.org/docs/master/generated/torch.nn.PReLU.html
+- MxNet - https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.nn.PReLU.html
+
+Example of calculations forward/backward can be found: https://bit.ly/2TXgvzT
+
+## Proposal
+
+1. Option A - create new dedicated primitive PReLU:
+
+``` cpp
+typedef struct {
+    /// The kind of primitive. Used for self-identifying the primitive
+    /// descriptor. Must be #dnnl_prelu.
+    dnnl_primitive_kind_t primitive_kind;
+    /// The kind of propagation. Possible values: #dnnl_forward_training,
+    /// #dnnl_forward_inference, #dnnl_backward, and #dnnl_backward_data.
+    dnnl_prop_kind_t prop_kind;
```

Comment:
i don't think we need to support `backward_data`, only `backward = backward_data & backward_weights`.